### PR TITLE
Compresses and Decompresses the index for ExDoc

### PR DESCRIPTION
**Pull request by jeregrine**
Successfully stores Elixir Hex Docs in Session storage

Screenshot 2023-07-11 at 10 49 44 AM

This looks more simple then it was since DecompressionStream/CompressionStream's are fairly new, but well supported outside of IE. https://caniuse.com/mdn-api_compressionstream

### CODE REVIEW

**Comment by jeregrine** [comment]
Partially fixes https://github.com/elixir-lang/ex_doc/issues/1732

File: assets/js/search-page.js [hunk 1]
```diff
@@ -74,13 +76,56 @@ function loadIndex () {
 
 function saveIndex (index) {
   try {
-    const serializedIndex = JSON.stringify(index)
-    sessionStorage.setItem(indexStorageKey(), serializedIndex)
+    const serializedIndex = compress(index).then((compressedIndex) => {
+      sessionStorage.setItem(indexStorageKey(), compressedIndex)
```
Comment (enewbury) [inline assets/js/search-page.js:79]: Do we want the try catch in the callback rather than the top level flow?

File: assets/js/search-page.js [hunk 2]
```diff
@@ -72,15 +73,56 @@ function loadIndex () {
...
+  const len = binary_string.length;
+  const bytes = new Uint8Array(new ArrayBuffer(len));
+  for (let i = 0; i < len; i++) {
+    bytes[i] = binary_string.charCodeAt(i);
+  }
+  return bytes;
+}
+
 function indexStorageKey () {
   return `index:${getProjectNameAndVersion()}`
```
Comment (josevalim) [inline assets/js/search-page.js:127]:
```suggestion
  return `idv2:${getProjectNameAndVersion()}`
```

Let's change the key just to ensure we don't accidentally load a past index through compression.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**